llvm/lib/CheerpUtils/CallConstructors.cpp

```diff
@@ -82,7 +87,20 @@ PreservedAnalyses CallConstructorsPass::run(llvm::Module &M, llvm::ModuleAnalysi
 				Argc = ConstantInt::get(ArgcTy, 0);
 				Argv = ConstantPointerNull::get(cast<PointerType>(ArgvTy));
 			}
-			ExitCode = Builder.CreateCall(Main->getFunctionType(), Main, { Argc, Argv });
+
```

Comment:
maybe we should assert/report error if main is in the "wrong" section?

Reply:
I don't understand what you mean, when is main in the "wrong" section? Do you mean when `__syscall_main_environ` and `main` are in different sections?

Reply:
if main is in wasm, but the target is cheerp-genericjs, then the libc is in genericjs, and you can't pass argv/env to main.
The other way around should work, but should be tested (main in genericjs, and target cheerp-wasm)

Reply:
I've currently made it so that when the target and main are not in the same sections that:
- argc will be `0`
- argv will be `nullptr`
- env will be a `char*` array with the first element a nullptr